ethcore/src/engines/null_engine.rs

```diff
@@ -14,103 +14,92 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
-use std::collections::BTreeMap;
-use util::Address;
-use builtin::Builtin;
-use block::{ExecutedBlock, IsBlock};
 use bigint::prelude::U256;
 use engines::Engine;
-use spec::CommonParams;
-use evm::Schedule;
-use header::BlockNumber;
-use error::Error;
-use state::CleanupMode;
-use trace::{Tracer, ExecutiveTracer, RewardType};
+use parity_machine::{Header, LiveBlock, WithBalances};
+
+/// Params for a null engine.
+#[derive(Clone, Default)]
+pub struct NullEngineParams {
+	/// base reward for a block.
+	pub block_reward: U256,
+}
+
+impl From<::ethjson::spec::NullEngineParams> for NullEngineParams {
+	fn from(p: ::ethjson::spec::NullEngineParams) -> Self {
+		NullEngineParams {
+			block_reward: p.block_reward.map_or_else(Default::default, Into::into),
+		}
+	}
+}
 
 /// An engine which does not provide any consensus mechanism and does not seal blocks.
-pub struct NullEngine {
-	params: CommonParams,
-	builtins: BTreeMap<Address, Builtin>,
+pub struct NullEngine<M> {
+	params: NullEngineParams,
+	machine: M,
 }
 
-impl NullEngine {
+impl<M> NullEngine<M> {
 	/// Returns new instance of NullEngine with default VM Factory
-	pub fn new(params: CommonParams, builtins: BTreeMap<Address, Builtin>) -> Self {
-		NullEngine{
+	pub fn new(params: NullEngineParams, machine: M) -> Self {
+		NullEngine {
 			params: params,
-			builtins: builtins,
+			machine: machine,
 		}
 	}
 }
 
-impl Default for NullEngine {
+impl<M: Default> Default for NullEngine<M> {
 	fn default() -> Self {
 		Self::new(Default::default(), Default::default())
 	}
 }
 
-impl Engine for NullEngine {
+impl<M: WithBalances> Engine<M> for NullEngine<M> {
 	fn name(&self) -> &str {
 		"NullEngine"
 	}
 
-	fn params(&self) -> &CommonParams {
-		&self.params
-	}
+	fn machine(&self) -> &M { &self.machine }
 
-	fn builtins(&self) -> &BTreeMap<Address, Builtin> {
-		&self.builtins
-	}
+	fn on_close_block(&self, block: &mut M::LiveBlock) -> Result<(), M::Error> {
```

Comment:
why not use `::engines::common::bestow_block_reward` here?

Reply:
since it handles uncle reward as well and there is a test that depends on that behavior